JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -149,51 +153,51 @@ module Microsoft.ApplicationInsights {
                     url = window.location && window.location.href || "";
                 }
 
-                var durationMs = 0;
-                // check if timing data is available
-                if (Telemetry.PageViewPerformance.checkPageLoad() !== undefined) {
-                    // compute current duration (navigation start to now) for the pageViewTelemetry
-                    var startTime = window.performance.timing.navigationStart;
-                    durationMs = Telemetry.PageViewPerformance.getDuration(startTime, +new Date);
-
-                    // poll for page load completion and send page view performance data when ready
-                    var handle = setInterval(() => {
-                        try {
-                            // abort this check if we have not finished loading after 1 minute
+                this.trackPageViewInternal(name, url, properties, measurements);
+            } catch (e) {
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.CRITICAL, "trackPageView failed, page view will not be collected: " + Util.dump(e));
+            }
+        }
+
+        private trackPageViewInternal(name?: string, url?: string, properties?: Object, measurements?: Object) {
+            var durationMs = 0;
+            // check if timing data is available
+            if (Telemetry.PageViewPerformance.isPerformanceTimingSupported()) {
+                // compute current duration (navigation start to now) for the pageViewTelemetry
+                var startTime = window.performance.timing.navigationStart;
+                durationMs = Telemetry.PageViewPerformance.getDuration(startTime, +new Date);
+
```

Comment:
I think for accurate measurements we should be collecting interval from navigatioStart to pageLoad event start.